nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/service/RenditionServiceImpl.java

```diff
@@ -221,6 +221,9 @@ protected StoredRendition storeRendition(DocumentModel sourceDocument, Rendition
             return null;
         }
         List<Blob> renderedBlobs = rendition.getBlobs();
+        if (renderedBlobs.isEmpty()) {
+            return null;
+        }
```

Comment:
according to the javadoc/comment `rendition.getBlobs()` can returns a null, then we can add control here to avoid NPE
`if (CollectionsUtils.isEmpty(renderedBlobs)) {
            return null;
        }
`